code/modules/events/nightmare.dm

```diff
@@ -0,0 +1,43 @@
+/datum/round_event_control/nightmare
+	name = "Spawn Nightmare"
+	typepath = /datum/round_event/ghost_role/nightmare
+	max_occurrences = 1
+	min_players = 20
+
+/datum/round_event/ghost_role/nightmare
+	minimum_required = 1
+	role_name = "nightmare"
+
+/datum/round_event/ghost_role/nightmare/spawn_role()
+	var/list/candidates = get_candidates("alien", null, ROLE_ALIEN)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected = pick_n_take(candidates)
+
+	var/datum/mind/player_mind = new /datum/mind(selected.key)
+	player_mind.active = 1
```

Comment:
TRUE

Reply:
mrty catching headcoder in their mistakes ayyy